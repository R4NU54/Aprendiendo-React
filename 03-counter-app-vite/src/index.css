:root {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

html,
body {
  color: rgba(255, 255, 255, 0.87);
  background-color: #242424;
  font-family: Helvetica, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  padding: 2rem;
  font-size: 1.3rem;
  border-radius: 0.5rem;
  transition: all 0.3s ease-in-out;

  &:hover {
    cursor: pointer;
    background-color: crimson;
    transition: all 0.3s ease-in-out;
    color: white;
  }
}
