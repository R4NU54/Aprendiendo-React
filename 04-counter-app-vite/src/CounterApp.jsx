import { useState } from 'react'

import PropTypes from 'prop-types'

export const CounterApp = ({ value }) => {
  const [Counter, setCounter] = useState(value)

  const handleSubstract = () => setCounter(Counter - 1)
  const handleReset = () => setCounter(value)
  const handleAdd = () => setCounter(Counter + 1)

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{Counter}</h2>
      <button onClick={handleSubstract}> -1 </button>
      <button onClick={handleReset}> Reset </button>
      <button onClick={handleAdd}> +1 </button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}
