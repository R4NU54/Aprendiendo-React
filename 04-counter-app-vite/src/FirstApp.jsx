import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <div>
      <h1 data-testid='test-title'> {title} </h1>
      <p>{subTitle}</p>
      <p>{name}</p>
    </div>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
  name: PropTypes.string,
};

FirstApp.defaultProps = {
  title: 'Titulo vacio',
  subTitle: 'Subtitulo vacio',
  name: 'Raúl Núñez',
};
