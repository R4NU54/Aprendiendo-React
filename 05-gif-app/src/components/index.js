export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';
export * from './GifHeader.jsx';
