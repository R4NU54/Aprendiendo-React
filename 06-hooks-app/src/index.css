html {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

body {
  padding: 2rem;
  min-width: 320px;
  min-height: 100vh;
}
