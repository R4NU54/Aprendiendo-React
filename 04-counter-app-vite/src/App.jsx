import { CounterApp } from './CounterApp';
// import { FirstApp } from './FirstApp';

function App() {
  return (
    <>
      <CounterApp value={0} />
      {/* <FirstApp title={'Curso React'} /> */}
    </>
  );
}

export default App;
