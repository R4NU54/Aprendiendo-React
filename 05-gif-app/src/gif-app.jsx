import { useState } from 'react';

import { AddCategory, GifGrid, GifHeader } from './components';

export const GifApp = () => {
  const [categories, setCategories] = useState(['Breaking Bad']);

  const handleAdd = (newCategory) => {
    if (categories.includes(newCategory)) return;

    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <GifHeader />
      <AddCategory onNewCategory={handleAdd} />

      {categories.map((category) => (
        <GifGrid
          key={category}
          category={category}
        />
      ))}
    </>
  );
};
