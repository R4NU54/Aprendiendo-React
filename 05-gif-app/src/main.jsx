import React from 'react';
import ReactDOM from 'react-dom/client';
import './styles.css';

import { GifApp } from './gif-app.jsx';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifApp />
  </React.StrictMode>
);
