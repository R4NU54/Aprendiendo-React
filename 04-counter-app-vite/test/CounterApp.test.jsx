import { fireEvent, render, screen } from '@testing-library/react';

import { CounterApp } from '../src/CounterApp';

describe('Test <CounterApp/>', () => {
  const inicialValue = 100;

  test('should match to snapshot', () => {
    const { container } = render(<CounterApp value={inicialValue} />);
    expect(container).toMatchSnapshot();
  });

  test('should show default value 100', () => {
    render(<CounterApp value={inicialValue} />);
    expect(screen.getByText(100)).toBeTruthy();
    // expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain('100');
  });

  test('should increment with button +1', () => {
    render(<CounterApp value={inicialValue} />);
    fireEvent.click(screen.getByText('+1'));
    expect(screen.getByText(101)).toBeTruthy();
  });

  test('should decrement with button -1', () => {
    render(<CounterApp value={inicialValue} />);
    fireEvent.click(screen.getByText('-1'));
    expect(screen.getByText(99)).toBeTruthy();
  });

  test('should reset with button reset', () => {
    render(<CounterApp value={355} />);
    fireEvent.click(screen.getByText('+1'));
    fireEvent.click(screen.getByText('+1'));
    fireEvent.click(screen.getByText('+1'));
    fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
    expect(screen.getByText(355)).toBeTruthy();
    screen.debug();
  });
});
