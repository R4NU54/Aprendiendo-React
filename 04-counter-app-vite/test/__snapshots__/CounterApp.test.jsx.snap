// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <CounterApp/> should match to snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button>
    -1
  </button>
  <button
    name="btn-reset"
  >
    Reset
  </button>
  <button>
    +1
  </button>
</div>
`;
